### DISCUSSION TITLE
Draw a cross centred at hovering mouse position over a TouchArea

### DISCUSSION THREAD

**Discussion by algorithmx** [slint-ui/slint#4867] [Q&A]
Dear experts,

I wish to draw lines with respect to the mouse cursor position. The simplest example would be, to draw a cross centred at the hovering mouse position, accessed via "mouse-x" and "mouse-y". But if I draw the line using "Path", the type of  point coordinates there should be int or float. I cannot use "mouse-x" and "mouse-y" because their types are length. How can I use them correctly? I cannot proceed on the experiment because this, so I don't even know whether it is possible to draw lines in the way I have chosen. 

Thanks in advance for any hints and replies! :-D

**Comment by Vadoola** [comment]
I believe you should just be able to divide the length by 1px and it will convert it to an int.

**Reply by Vadoola** [reply]
I would try and create a small example on slintpad, but I've never managed to get slintpad to work on mobile.

**Comment by Vadoola** [comment]
[Here is a quickly thrown together example](https://slintpad.com/?snippet=import+%7B+AboutSlint%2C+Button%2C+VerticalBox+%7D+from+%22std-widgets.slint%22%3B%0Aexport+component+Demo+%7B%0A++++Rectangle+%7B%0A++++++++background%3A+white%3B%0A++++++++height%3A+400px%3B%0A++++++++width%3A+400px%3B%0A++++++++ta+%3A%3D+TouchArea+%7B%0A++++++++%0A++++++++%7D%0A%0A++++++++Path+%7B%0A++++++++++++height%3A+10%25%3B%0A++++++++++++width%3A+10%25%3B%0A++++++++++++x%3A+ta.mouse-x%3B%0A++++++++++++y%3A+ta.mouse-y%3B%0A++++++++++++viewbox-height%3A+self.height%2F1px%3B%0A++++++++++++viewbox-width%3A+self.width%2F1px%3B%0A++++++++++++stroke-width%3A+2px%3B%0A++++++++++++stroke%3A+red%3B%0A%0A++++++++++++MoveTo+%7B%0A++++++++++++++++x%3A+5%3B%0A++++++++++++++++y%3A+0%3B%0A++++++++++++%7D%0A++++++++++++LineTo+%7B%0A++++++++++++++++x%3A+5%3B%0A++++++++++++++++y%3A+10%3B%0A++++++++++++%7D%0A++++++++++++MoveTo+%7B%0A++++++++++++++++x%3A+0%3B%0A++++++++++++++++y%3A+5%3B%0A++++++++++++%7D%0A++++++++++++LineTo+%7B%0A++++++++++++++++x%3A+10%3B%0A++++++++++++++++y%3A+5%3B%0A++++++++++++%7D%0A%0A++++++++%7D%0A++++%7D%0A%7D%0A&style=native). This draws the path within it's viewbox and then just moves the path around based on the mouse position
